Home work1

Shtuka1993 opened the pull request:
Home work one

yshtokaylo committed:
Home Work One

yshtokaylo committed:
Refactored code

yderevianko-cartera commented on src/01-basics.php:
```diff
@@ -12,6 +12,32 @@
  */
 function getMinuteQuarter(int $minute)
 {
+    define("Q1","first");
```
По перше константи не рекомендовано дефайнити всереди функції так як це спричиняє помилки типу:

GetMinuteQuarterTest::testPositive with data set #1 (5, 'first')
Constant Q1 already defined
/vagrant/www/phplab-tasks/src/01-basics.php:15
/vagrant/www/phplab-tasks/tests/GetMinuteQuarterTest.php:12

І взагалі є суть створбвати константи якщо вони будуть використовуватися принаймі в 2-ох різних місцях, а в цій домашці вони можуть просто бути замінени на стрічки в середині switch.

Shtuka1993 replied:
Дякую за рекомендації. Прибрав константи. В принципі в моєму коді "fourth" використано двічі, то ж думав що можна використати константи.

yderevianko-cartera commented on src/01-basics.php:
```diff
@@ -12,6 +12,32 @@
+        if($minute == 0) {
+            return Q4;
+        }
+        $quarter = (int)(($minute-1)/15)+1;
```
PSR-1\PSR-2 
 $quarter = (int)(($minute - 1) / 15) + 1;

Shtuka1993 replied:
Дякую. Працюю над цим. Виправив.

yderevianko-cartera commented on src/01-basics.php:
```diff
@@ -42,4 +73,18 @@ function isLeapYear(int $year)
+    $s2= 0;
+
+    if (strlen($input) === 6) {
+        for( $i=0; $i<3; $i++ ) {
```
PSR
for ($i = 0; $i < 3; $i++) {

Shtuka1993 replied:
Дякую. Працюю над цим. Виправив.

yderevianko-cartera commented on src/01-basics.php:
```diff
@@ -42,4 +73,18 @@ function isLeapYear(int $year)
+    $s1 = 0;
+    $s2= 0;
+
+    if (strlen($input) === 6) {
```
Tут все працює, але можна ще додати конвертацію типу щоб php сам не вгадував.
Також можна зберегти собі розмір стрічки і зробити логіку трохи більш динамічною.

$n = strlen($input);
...
for ($i = 0; $i < $n / 2; $i++) {
    $s1 += (int)$input[$i];
    $s2 += (int)$input[$n - 1 - $i];

Shtuka1993 replied:
Оновив.

yshtokaylo committed:
Fixes on comments in pull request.
Removed constants, updated to PSR standards, added some dynamic behavior for the third function.

yderevianko-cartera commented on src/01-basics.php:
```diff
@@ -12,6 +12,32 @@
 {
+    if (($minute >= 0) AND ($minute <= 60)) {
+        if($minute == 0) {
+
```
Краще не використовувати return в свічі. Краще присвоювати результат якійсь змінній і в кінці зробити один return.
І не потрібно пустий рядок перед return якщо він перший рядок в блоці.

Shtuka1993 replied:
Оновив.

yshtokaylo committed:
Fixes on comments in pull request. Removed multiple returns in switch.

yderevianko-cartera commented on src/01-basics.php:
```diff
@@ -12,6 +12,30 @@
+    $result = '';
+    if (($minute >= 0) AND ($minute <= 60)) {
+        if($minute == 0) {
+            $result = "fourth";
```
тут мабуть потрібно одразу return 'fourth'

Shtuka1993 replied:
Дякую. Виправив.

yshtokaylo committed:
Fix on comment in pull request. Added return statement in if when minute is 0

yshtokaylo committed:
Refactored code.
